### PR TITLE
AWS Auth Method

### PR DESCRIPTION
**Pull request by drassaby**
Not actually integration tested yet, but unit tests are present ensuring AWS authentication is handled properly on this end

Version bump -> `0.2.0-SNAPSHOT`

### CODE REVIEW

File: src/amperity/gocd/secret/vault/plugin.clj [hunk 1]
```diff
@@ -150,12 +155,18 @@
   - `client` The Vault Client you wish to authenticate
   - `inputs` A map containing the user inputted settings for the plugin"
   [client inputs]
-  (case (keyword (:auth_method inputs))
+  (case (keyword (str/join "-" (str/split (str/lower-case (:auth_method inputs)) #" ")))
```
Comment (pyncc) [inline src/amperity/gocd/secret/vault/plugin.clj]: this is pretty gnarly, could just match on the string you get instead of parsing to a keyword.

**Review by pyncc — APPROVED** [review]
LGTM

File: src/amperity/gocd/secret/vault/plugin.clj (hunk 1 above)
Comment (drassaby) [reply src/amperity/gocd/secret/vault/plugin.clj]: Yeah, I'm gonna go back over and make a few changes (I'm going to change here while I'm at it). I've been spending a while trying to figure out why I can't test using passed in credentials. Will lyk when I find out more

Comment (drassaby) [reply src/amperity/gocd/secret/vault/plugin.clj]: Got it working 👍

**Review by pyncc — APPROVED** [review]
Nice polish and great tests.

**Pull request merged**